library/core/src/main/java/com/google/android/exoplayer2/extractor/flv/FlvExtractor.java

```diff
@@ -110,6 +110,7 @@ public boolean sniff(ExtractorInput input) throws IOException, InterruptedExcept
     input.peekFully(scratch.data, 0, 4);
     scratch.setPosition(0);
     int dataOffset = scratch.readInt();
+    if (dataOffset > 9) dataOffset = 9;
```

Comment:
@ojw28 Do you have any suggestions for best fixing this. According to flv specifications, 9 should be the value in both flv0 and flv1. There were never any newer flv versions released. 

https://www.adobe.com/content/dam/Adobe/en/devnet/flv/pdfs/video_file_format_spec_v10.pdf

In our code, we had previously set an arbitrary number of 200 as a limit ```(dataOffset > 200)```. It was a quick win when we were plagued by OOM exceptions that can be caused from the data offset being huge. I hate to throw a random number in there, but there should be some sort of max number that is agreed upon.